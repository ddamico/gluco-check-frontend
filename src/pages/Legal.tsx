import React from "react";
import { Trans, useTranslation } from "react-i18next";
import { Container, Grid, makeStyles, Typography } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
  root: {},
  contentGrid: {
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(2),
    "& p, & h2, & h3": {
      marginBottom: theme.spacing(1),
    },
    "& section": {
      marginBottom: theme.spacing(3),
    },
  },
}));

function Legal() {
  const classes = useStyles();
  const { t } = useTranslation();

  return (
    <>
      <Grid
        container
        direction="column"
        justify="center"
        alignItems="center"
        className={classes.contentGrid}
      >
        <Grid item>
          <Container component="section" maxWidth="md">
            <Typography variant="h5" component="h2">
              {t("termsOfService.title")}
            </Typography>
            <Typography variant="body1" component="h3">
              {t("termsOfService.subtitle")}
            </Typography>
            <Typography variant="body1" component="p">
              {t("termsOfService.p1")}
            </Typography>
            <Typography variant="body1" component="p">
              {t("termsOfService.p2")}
            </Typography>
            <Typography variant="body1" component="p">
              {t("termsOfService.p3")}
            </Typography>
          </Container>
          <Container component="section" maxWidth="md">
            <Typography variant="h5" component="h2">
              {t("medicalDisclaimer.title")}
            </Typography>
            <Typography variant="body1" component="p">
              {t("medicalDisclaimer.p1")}
            </Typography>
          </Container>
          <Container component="section" maxWidth="md">
            <Typography variant="h5" component="h2">
              {t("privacyPolicy.title")}
            </Typography>
            <Typography variant="body1" component="h3">
              {t("privacyPolicy.p1.title")}
            </Typography>
            <Typography variant="body1" component="p">
              {t("privacyPolicy.p1.body")}
            </Typography>
            <Typography variant="body1" component="h3">
              {t("privacyPolicy.p2.title")}
            </Typography>
            <Typography variant="body1" component="p">
              {t("privacyPolicy.p2.body")}
            </Typography>
            <Typography variant="body1" component="h3">
              {t("privacyPolicy.p3.title")}
            </Typography>
            <Typography variant="body1" component="p">
              <Trans key="privacyPolicy.p3.body">
                Your email address or Nightscout URL will <strong>never</strong>{" "}
                be shared with a third party. I just need it so I know which
                Nightscout site to check when you invoke the Action. Data from
                your Nightscout website is not saved, not even to log files.
              </Trans>
            </Typography>
            <Typography variant="body1" component="h3">
              {t("privacyPolicy.p4.title")}
            </Typography>
            <Typography variant="body1" component="p">
              {t("privacyPolicy.p4.body")}
            </Typography>
            <Typography variant="body1" component="h3">
              {t("privacyPolicy.p5.title")}
            </Typography>
            <Typography variant="body1" component="p">
              <Trans key="privacyPolicy.p5.body">
                After you sign in to this site, some anonymous statistics about
                your visit are collected using a cookie (Google Analytics). The
                cookie is not enabled until after you've accepted this Privacy
                Policy. I'm using Analytics to better understand how people are
                using this website, how they found it, and which demographic
                they're in. This only applies to this site, not to usage of the
                Assistant Action itself. If you'd like to opt-out of the Google
                Analytics cookie, you can do so by visiting{" "}
                <a href="https://tools.google.com/dlpage/gaoptout">
                  https://tools.google.com/dlpage/gaoptout/
                </a>
              </Trans>
            </Typography>
          </Container>
        </Grid>
      </Grid>
    </>
  );
}

export default Legal;
