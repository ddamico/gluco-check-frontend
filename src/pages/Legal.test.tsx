import React from "react";
import { cleanup, render } from "@testing-library/react";
import { axe } from "jest-axe";
import Legal from "./Legal";

jest.mock("react-i18next", () => ({
  useTranslation: () => {
    return {
      t: jest.fn().mockImplementation((i) => {
        return i;
      }),
    };
  },
  Trans: function (props: any) {
    return <span>{props.i18nKey}</span>;
  },
}));

afterEach(cleanup);

describe("Legal page", () => {
  it("renders the component", () => {
    const { container } = render(<Legal />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("has no axe violations", async () => {
    const { container } = render(<Legal />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
