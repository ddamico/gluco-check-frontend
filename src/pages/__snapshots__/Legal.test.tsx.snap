// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legal page renders the component 1`] = `
<div
  class="MuiGrid-root makeStyles-contentGrid-2 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <section
      class="MuiContainer-root MuiContainer-maxWidthMd"
      data-testid="tos"
      id="tos"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5"
      >
        termsOfService.title
      </h2>
      <h3
        class="MuiTypography-root MuiTypography-body1"
      >
        termsOfService.subtitle
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        termsOfService.p1
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        termsOfService.p2
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        termsOfService.p3
      </p>
    </section>
    <section
      class="MuiContainer-root MuiContainer-maxWidthMd"
      data-testid="medical"
      id="medical"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5"
      >
        medicalDisclaimer.title
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        medicalDisclaimer.p1
      </p>
    </section>
    <section
      class="MuiContainer-root MuiContainer-maxWidthMd"
      data-testid="privacy"
      id="privacy"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5"
      >
        privacyPolicy.title
      </h2>
      <h3
        class="MuiTypography-root MuiTypography-h6"
      >
        privacyPolicy.p1.title
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        privacyPolicy.p1.body
      </p>
      <h3
        class="MuiTypography-root MuiTypography-h6"
      >
        privacyPolicy.p2.title
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        privacyPolicy.p2.body
      </p>
      <h3
        class="MuiTypography-root MuiTypography-h6"
      >
        privacyPolicy.p3.title
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        <span>
          privacyPolicy.p3.body
        </span>
      </p>
      <h3
        class="MuiTypography-root MuiTypography-h6"
      >
        privacyPolicy.p4.title
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        privacyPolicy.p4.body
      </p>
      <h3
        class="MuiTypography-root MuiTypography-h6"
      >
        privacyPolicy.p5.title
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        <span>
          privacyPolicy.p5.body
        </span>
      </p>
    </section>
  </div>
</div>
`;
